endpoint.go

```diff
@@ -328,6 +427,7 @@ func (ep *endpoint) Leave(containerID string, options ...EndpointOption) error {
 	}
 	ep.container = nil
 	ep.Unlock()
+	ep.network.ctrlr.addEndpointToStore(ep)
```

Comment:
This should be called after we get the `ctrlr` pointer while holding the network lock and then we invoke `addEndpointToStore`